presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoDatabaseMetaData.java

```diff
@@ -1247,29 +1238,41 @@ public int getResultSetHoldability()
     public int getDatabaseMajorVersion()
             throws SQLException
     {
-        // TODO: get version from server
-        return PrestoDriver.VERSION_MAJOR;
+        return getDatabaseVersionPart(0);
     }
 
     @Override
     public int getDatabaseMinorVersion()
             throws SQLException
     {
-        return PrestoDriver.VERSION_MINOR;
+        return getDatabaseVersionPart(1);
+    }
+
+    private int getDatabaseVersionPart(int part)
+            throws SQLException
+    {
+        String version = getDatabaseProductVersion();
+        List<String> parts = Splitter.on('.').limit(3).splitToList(version);
+        try {
+            return parseInt(parts.get(part));
+        }
+        catch (IndexOutOfBoundsException | NumberFormatException e) {
+            return 0;
```

Comment:
2 minor comments/questions:

1. Will this return 0 for every snapshot version when `getDatabaseMinorVersion()` is called? If so, that seems like a pain for us or anybody doing testing.
2. For end users, it seems likely that the only reason you'd hit this is because the presto versioning scheme has radically changed and the driver version parsing code is out of date. It seems like a good bet that the user needs to update the driver. How about throwing an SQLException containing the full version string like "Driver version x.y could not parse server version string '%s'. Try updating the driver"

Please feel free to ignore if this is spec-compliant behavior as-is.